### PR TITLE
feat(ChatFeedback): add a four-point rating flow for conversational surfaces

### PR DESCRIPTION
**Pull request by annamalailakshmann**
## Description

A rating flow for chat surfaces: ask a question, take a mood on a four-point scale, follow up with tags, confirm, leave. Blade owns the machinery — radio-group semantics, per-point tooltips, hit targets, selected state. **The artwork is the consumer's:** `moodIcons` is required, because Blade ships no set for this scale yet.

📄 **[Full state review and API contract](https://github.com/razorpay/blade/blob/feat/chat-feedback/packages/blade/src/components/ChatFeedback/_decisions/decisions.md)** — states, transitions, invariants, alternatives considered, and open questions.

## Changes

- `ChatFeedback` (web) + a native stub that throws until implemented
- `ChatFeedbackCheck` — the closing mark, a filled positive disc with a check, composed from `CheckIcon` and tokens
- Per-mood closing copy: the unhappy end is acknowledged, not thanked for a lovely contribution
- Thank-you step held 1.3s before dismissing
- `onTagsChange` / `controlsRef` / `isSubmitHidden` — the seam a host uses to collect free text in an input of its own
- KitchenSink story, docs stories, interaction tests, unit tests including accessibility

## Two decisions worth a reviewer's eye

**The selected state lives on the button, not the glyph.** It used to be three cues, all on the icon — recolour, lift, and swapping in a filled twin. With consumer artwork that may be untintable and has no twin, two of the three silently do nothing, leaving a 12% scale as the only sign a rating registered.

**32px targets around 20px glyphs.** Butted together, 20 + 12 gives the 32px pitch. An earlier 24px glyph left 4px of padding, so a tap a couple of pixels wide of a face recorded the *adjacent* rating — a wrong answer, not a near miss.

## Breaking within this unreleased component

The free-text `comment` step and its **Add more feedback** link are removed, with `addCommentLabel` and `commentPlaceholder`. Free text is now the host's to collect; `comment` folds it into the submit payload. `ChatFeedbackStep` loses `'comment'`.

Consequence: a surface without a composer can no longer collect free text at all. Easy to reinstate if needed.

## Additional Information

Third of a stack of four; based on #3884 for `motion.easing.settle`. `ChatComposer` and a candidate icon set exist locally and are deliberately **not** in this PR.

`controlsRef` is assigned during render so the handle cannot go stale — at odds with React's ref guidance and worth discussing for concurrent rendering. Noted as an open question in the contract.

## Component Checklist

- [x] Update Component Status Page
- [ ] Perform Manual Testing in Other Browsers
- [x] Add KitchenSink Story
- [x] Add Interaction Tests (if applicable)
- [x] Add changeset

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+      display="flex"
+      flexDirection="column"
+      alignItems="stretch"
+      width={isFullWidth ? '100%' : 'auto'}
+      {...metaAttribute({ name: MetaConstants.ChatFeedback, testID })}
+      {...makeAnalyticsAttribute(rest)}
+      {...getStyledProps(rest)}
+    >
+      {/* Keyed on step so each arrival replays the entrance. */}
+      <BaseMotionBox key={step} motionVariants={stepInVariants}>
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx:326]: 🟠 **[MAJOR]** · _code-quality-critique_ · confidence: 9/10

**Problem:** BaseMotionBox is keyed on `step` to replay the entrance animation, but it is not wrapped in <AnimatePresence>. Framer-motion requires AnimatePresence to defer unmounting until the exit animation completes. Without it, the old step is instantly removed from the DOM when the key changes, so the `exit` variant defined in stepInVariants never plays. The exit variant is dead code. ChatInput.web.tsx correctly wraps its BaseMotionBox in <AnimatePresence>.

**Suggestion:** Wrap the keyed BaseMotionBox in <AnimatePresence mode='wait'>.

File: packages/blade/src/components/ChatFeedback/ChatFeedbackMoodScale.web.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+}: ChatFeedbackMoodScaleProps): React.ReactElement => {
+  const { theme } = useTheme();
+
+  return (
+    <BaseBox
+      display="flex"
+      flexDirection="row"
+      alignItems="center"
+      flexShrink={0}
+      role="radiogroup"
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedbackMoodScale.web.tsx:216]: 🟠 **[MAJOR]** · _code-quality-critique_ · confidence: 8/10

**Problem:** The mood scale uses role="radiogroup" with <button role="radio"> but does not implement the WAI-ARIA keyboard interaction pattern (arrow keys, roving tabindex). Every button is an independent tab stop.

**Suggestion:** Add roving tabindex and arrow key handler, or use native <input type="radio">.

File: packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        </BaseBox>
+        {hasSubmittedComment ? null : (
+          <Link variant="button" size="small" color="neutral" onClick={openComment}>
+            {addCommentLabel}
+          </Link>
+        )}
+      </BaseBox>
+    ),
+
+    comment: () => (
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx]: 🟠 **[MAJOR]** · _code-quality-critique_ · confidence: 8/10

**Problem:** The comment step has no back, cancel, or close affordance. A user who clicks 'Add more feedback' and changes their mind is trapped.

**Suggestion:** Add a cancel/close affordance and Escape-key handler.

File: packages/blade/src/components/ChatFeedback/index.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export type {
+  ChatFeedbackProps,
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/index.ts:3]: 🟠 **[MAJOR]** · _api-decision-critique_ · confidence: 8/10

**Problem:** `ChatFeedbackMoodIcons` is exported from types.ts but not re-exported from index.ts. Consumers cannot type their icon set with a named type, inconsistent with exporting ChatFeedbackMood and ChatFeedbackMoodConfig.

**Suggestion:** Add `ChatFeedbackMoodIcons` to the type export list in index.ts.

**Review by kamaleshs-bridge4 — COMMENTED** [review]
## ✨ Agentic PR Review ✨

### UI Review

✅ 3 passed · ❌ 5 failed

| Check | Problem | Screenshot |
|-------|---------|------------|
| [ChatFeedback Default story](http://localhost:9009/?path=/story/components-chatfeedback--default) | The Default story omits the required moodIcons prop, causing a runtime crash in ChatFeedbackMoodScale.web.tsx:152 (Cannot read properties of undefined (reading 'very-dissatisfied')). The story renders a Storybook error page instead of the component. | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-chatfeedback--default-1.png"></a> |
| [ChatFeedback Without auto dismiss story](http://localhost:9009/?path=/story/components-chatfeedback--without-auto-dismiss) | The 'Without auto dismiss' story omits the required moodIcons prop, causing the same runtime crash as the Default story. | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-chatfeedback--without-auto-dismiss-2.png"></a> |
| [ChatFeedback Custom mood config story](http://localhost:9009/?path=/story/components-chatfeedback--custom-mood-config) | The 'Custom mood config' story provides a moodConfig override but omits the required moodIcons prop, causing the same runtime crash. | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-chatfeedback--custom-mood-config-3.png"></a> |
| [ChatFeedback KitchenSink story](http://localhost:9009/?path=/story/components-kitchensink-chatfeedback--chat-feedback) | The KitchenSink story composes the three broken stories from ChatFeedback.stories.tsx (none of which pass moodIcons), causing a cascading React error: 'Rendered more hooks than during the previous render.' The entire KitchenSink page fails to render. | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-kitchensink-chatfeedback--chat-feedback-7.png"></a> |
| [ChatFeedback Docs page](http://localhost:9009/?path=/docs/components-chatfeedback--docs) | The Docs page's embedded Example canvas crashes because it renders the Default story, which omits the required moodIcons prop. The error message 'Cannot read properties of undefined (reading very-dissatisfied)' is displayed in place of the live component preview. | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/story-7-8.png"></a> |

<details>
<summary>Passing checks (3)</summary>

| Check | Screenshot |
|-------|------------|
| ✅ [ChatFeedback Interaction Tests — Hit targets resolve to the right mood](http://localhost:9009/?path=/story/components-chatfeedback-chatfeedback-interaction-tests--hit-targets-resolve-to-the-right-mood) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-chatfeedback-chatfeedback-interaction-tests--hit-targets-resolve-to-the-right-mood-4.png"></a> |
| ✅ [ChatFeedback Interaction Tests — Selection is visible with untintable artwork](http://localhost:9009/?path=/story/components-chatfeedback-chatfeedback-interaction-tests--selection-is-visible-with-untintable-artwork) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-chatfeedback-chatfeedback-interaction-tests--selection-is-visible-with-untintable-artwork-5.png"></a> |
| ✅ [ChatFeedback Interaction Tests — The flow advances and returns](http://localhost:9009/?path=/story/components-chatfeedback-chatfeedback-interaction-tests--the-flow-advances-and-returns) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/02-36-20/components-chatfeedback-chatfeedback-interaction-tests--the-flow-advances-and-returns-6.png"></a> |

</details>

### Usage

```jsx
import { ChatFeedback } from '@razorpay/blade/components';

<ChatFeedback
  moodIcons={{
    'very-dissatisfied': <span>😢</span>,
    'dissatisfied': <span>😕</span>,
    'satisfied': <span>🙂</span>,
    'very-satisfied': <span>😍</span>,
  }}
  onSubmit={(payload) => console.log(payload)}
  onDismiss={() => setShow(false)}
/>
```

File: packages/blade/src/components/ChatFeedback/docs/ChatFeedback.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        <Button variant="tertiary" size="small" onClick={() => setShow(true)}>
+          Show feedback again
+        </Button>
+      )}
+    </Box>
+  );
+};
+
+export const Default = ChatFeedbackTemplate.bind({});
+Default.args = { question: "How's Ray doing so far?" };
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/docs/ChatFeedback.stories.tsx:87]: 🟠 **[MAJOR]** · _api-decision-critique_ · confidence: 9/10

**Problem:** All three story definitions omit the required `moodIcons` prop from their `.args`, causing a runtime crash. The Sandbox usage example also omits `moodIcons`.

**Suggestion:** Add `moodIcons` to each story's args and update the Sandbox example.

File: packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+    goBackToMood,
+    openComment,
+    submitComment,
+  } = useChatFeedback({ moodConfig, onMoodSelect, onSubmit, onDismiss, autoDismiss });
+
+  const [comment, setComment] = React.useState('');
+
+  // Steps arrive on the settle curve. The exit is a plain fade — the incoming step is the
+  // thing worth watching, so the outgoing one should get out of the way quickly.
+  const stepInVariants: MotionVariantsType = {
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx:164]: 🔵 **[MINOR]** · _code-quality-critique_ · confidence: 6/10

**Problem:** stepInVariants gets a new identity on every render, causing unnecessary animation re-initialisation.

**Suggestion:** Wrap in useMemo or move outside component.

File: packages/blade/src/components/ChatFeedback/__tests__/ChatFeedback.web.test.tsx [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    await userEvent.click(getByLabelText('Good'));
+    await userEvent.click(await findByText('Helpful'));
+    await userEvent.click(await findByRole('button', { name: 'Submit feedback' }));
+
+    await waitFor(() => expect(onDismiss).toHaveBeenCalledTimes(1), { timeout: 4000 });
+  });
+
+  // Synchronous by design: fake timers drive the whole assertion, so there is nothing to await.
+  it('should not call onDismiss when autoDismiss is off', () => {
+    jest.useFakeTimers();
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/__tests__/ChatFeedback.web.test.tsx:125]: 🔵 **[MINOR]** · _code-quality-critique_ · confidence: 7/10

**Problem:** jest.useFakeTimers() in test body can leak if assertions throw before jest.useRealTimers().

**Suggestion:** Use beforeEach/afterEach for fake timer setup.

File: packages/blade/src/components/ChatFeedback/ChatFeedbackCheck.web.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ * this size reads as one more control rather than as a result. The filled disc is the final one.
+ *
+ * 16px sits at about the cap height of the line beside it, so the mark reads as punctuation on
+ * that sentence rather than competing with it; an earlier 20px disc was the loudest thing on a
+ * strip whose whole job at that moment is to leave. The check is sized independently — Blade's
+ * icon scale is xsmall 8, small 12, medium 16 — so `small` fills the disc to a 2px ring. At
+ * `xsmall` the 8px tick floated in a green field and read as a dot: it is the ring, not the disc,
+ * that makes it read as a check.
+ */
+const DISC_SIZE = { small: '16px', medium: '40px' } as const;
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedbackCheck.web.tsx:34]: 🔵 **[MINOR]** · _code-quality-critique_ · confidence: 6/10

**Problem:** The 'medium' size is defined but never used anywhere. Dead code.

**Suggestion:** Remove or document planned usage.

File: packages/blade/src/components/ChatFeedback/useChatFeedback.ts [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+  const submitComment = React.useCallback(
+    (comment: string) => {
+      if (!selectedMood) return;
+      onSubmit?.({ mood: selectedMood, tags: selectedTags, comment });
+      setHasSubmittedComment(true);
+      clearTimers();
+      setStep('thanks');
+      if (!autoDismiss) return;
+      schedule(() => onDismiss?.(), theme.motion.delay.xgentle);
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/useChatFeedback.ts]: 🙏🏻 **[NEEDS CLARIFICATION]** · _code-quality-critique_ · confidence: 4/10

Auto-dismiss delay after comment (960ms) is shorter than after tags (2000ms). Is this intentional?

File: packages/blade/src/components/ChatFeedback/__tests__/ChatFeedback.web.test.tsx [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+
+    await userEvent.click(getByLabelText('Good'));
+    const submit = await findByRole('button', { name: 'Submit feedback' });
+    expect(submit).toBeDisabled();
+
+    await userEvent.click(getByText('Helpful'));
+    expect(submit).toBeEnabled();
+  });
+
+  it('should submit the mood and selected tags', async () => {
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/__tests__/ChatFeedback.web.test.tsx:58]: 🔵 **[MINOR]** · _code-quality-critique_ · confidence: 6/10

**Problem:** No test for submitting tags with empty selection, no test for isFullWidth={false}, no test for rapid double-mood-selection.

**Suggestion:** Add tests for these edge cases.

File: packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+            />
+          </BaseBox>
+          <Button
+            type="submit"
+            variant="primary"
+            color="primary"
+            size="small"
+            isDisabled={isDisabled || !comment.trim()}
+          >
+            Send
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedback.web.tsx]: 🔵 **[MINOR]** · _api-decision-critique_ · confidence: 7/10

**Problem:** Comment submit button label 'Send' is hardcoded, no commentSubmitLabel prop for i18n.

**Suggestion:** Add a `commentSubmitLabel` prop with @default 'Send'.

File: packages/blade/src/components/ChatFeedback/types.ts [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+   * @default 'Anything else you would like to share?'
+   */
+  commentPlaceholder?: string;
+
+  /**
+   * Whether the flow dismisses itself shortly after the thank-you step.
+   * When false, `onDismiss` is never fired automatically and you control removal.
+   * @default true
+   */
+  autoDismiss?: boolean;
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/types.ts:115]: 🔵 **[MINOR]** · _api-decision-critique_ · confidence: 6/10

**Problem:** `autoDismiss` breaks the `is`-prefix convention used by isFullWidth and isDisabled on the same interface.

**Suggestion:** Consider renaming to `isAutoDismiss` for consistency.

File: packages/blade/src/components/ChatFeedback/types.ts [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+   */
+  onMoodSelect?: ({ mood }: { mood: ChatFeedbackMood }) => void;
+
+  /**
+   * Callback fired when the user submits their feedback.
+   *
+   * Fires once when tags are submitted, and again with `comment` populated if the user
+   * goes on to add a free-text follow-up.
+   */
+  onSubmit?: (payload: ChatFeedbackSubmitPayload) => void;
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/types.ts:95]: 🙏🏻 **[NEEDS CLARIFICATION]** · _api-decision-critique_ · confidence: 4/10

`onSubmit` fires twice when user adds a comment but payload has no discriminator. Is checking `payload.comment !== undefined` the intended signal?

File: packages/blade/src/components/ChatFeedback/ChatFeedbackMoodScale.web.tsx [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+      aria-label="Rate this experience"
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/ChatFeedbackMoodScale.web.tsx:217]: 🔵 **[MINOR]** · _api-decision-critique_ · confidence: 6/10

**Problem:** Accessibility labels are hardcoded (Rate this experience, Back to rating, Feedback tags, Submit feedback). No props for i18n.

**Suggestion:** Consider adding accessibilityLabels prop or document they are intentionally fixed.

File: packages/blade/src/utils/metaAttribute/metaConstants.ts [hunk 15]
```diff
@@ -29,6 +29,8 @@ export const MetaConstants = {
   BottomNav: 'bottomnav',
   BottomNavItem: 'bottomnav-item',
   Carousel: 'carousel',
+  ChatComposer: 'chat-composer',
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/utils/metaAttribute/metaConstants.ts:32]: 🔵 **[MINOR]** · _api-decision-critique_ · confidence: 7/10

**Problem:** `ChatComposer` is added to MetaConstants but no ChatComposer component exists in this PR. Appears to be unrelated.

**Suggestion:** Remove ChatComposer or add a comment.

File: packages/blade/src/components/ChatFeedback/types.ts [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+  /** Replaces the selected tags. */
+  setTags: (tags: string[]) => void;
+};
+
+type ChatFeedbackSubmitPayload = {
+  mood: ChatFeedbackMood;
+  /** Tags the user selected. Empty when they submitted without picking any. */
+  tags: string[];
+  /** Free-text follow-up, present only when the user chose to add more. */
+  comment?: string;
```
Comment (kamaleshs-bridge4) [inline packages/blade/src/components/ChatFeedback/types.ts:41]: 🟠 **[MAJOR]** · _api-decision-critique_ · confidence: 8/10

**Problem:** `ChatFeedbackSubmitPayload.comment` is dead API surface. The field is declared (and exported in index.ts), the `onSubmit` JSDoc (types.ts:92) says it 'fires once when tags are submitted, and again with `comment` populated if the user goes on to add a free-text follow-up', and both the changeset and decisions.md state `ChatFeedbackProps.comment` folds free-text into the payload — but `ChatFeedbackProps` has no `comment` prop and `useChatFeedback.submitTags` (useChatFeedback.ts:104) calls `onSubmit?.({ mood, tags })` with no `comment`, so the field is always `undefined`. The exported type and docs promise a value the implementation never delivers.

**Suggestion:** Either wire a `comment?: string` prop on `ChatFeedbackProps` through to the payload (so a host collecting free text can fold it in), or remove `comment` from `ChatFeedbackSubmitPayload` and correct the `onSubmit` JSDoc, changeset and decisions.md to match. Don't ship a public type field that can never be populated.

**Review by kamaleshs-bridge4 — COMMENTED** [review]
## ✨ Agentic PR Review ✨

### UI Review

✅ 7 passed

<details>
<summary>Passing checks (7)</summary>

| Check | Screenshot |
|-------|------------|
| ✅ [ChatFeedback mood step](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-chatfeedback--default) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-chatfeedback--default-1.png"></a> |
| ✅ [ChatFeedback tags step](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-chatfeedback--default) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-chatfeedback--default-2.png"></a> |
| ✅ [ChatFeedback thanks step](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-chatfeedback--without-auto-dismiss) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-chatfeedback--without-auto-dismiss-3.png"></a> |
| ✅ [ChatFeedback without auto dismiss](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-chatfeedback--without-auto-dismiss) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-chatfeedback--without-auto-dismiss-4.png"></a> |
| ✅ [ChatFeedback custom mood config](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-chatfeedback--custom-mood-config) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-chatfeedback--custom-mood-config-6.png"></a> |
| ✅ [ChatFeedback back navigation](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-chatfeedback--custom-mood-config) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-chatfeedback--custom-mood-config-6.png"></a> |
| ✅ [ChatFeedback kitchen sink](https://61c19ee8d3d282003ac1d81c-hibenwauwn.chromatic.com/?path=/story/components-kitchensink-chatfeedback--chat-feedback) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/08-05-41/components-kitchensink-chatfeedback--chat-feedback-7.png"></a> |

</details>

### Usage

```jsx
import { ChatFeedback } from '@razorpay/blade/components';

const moodIcons = {
  'very-dissatisfied': <span>😢</span>,
  dissatisfied: <span>😕</span>,
  satisfied: <span>🙂</span>,
  'very-satisfied': <span>😍</span>,
};

<ChatFeedback
  question="How's Ray doing so far?"
  moodIcons={moodIcons}
  onSubmit={({ mood, tags }) => record(mood, tags)}
  onDismiss={() => setShowFeedback(false)}
/>
```

**Review by kamaleshs-bridge4 — COMMENTED** [review]
## ✨ Agentic PR Review ✨

### UI Review

✅ 7 passed

<details>
<summary>Passing checks (7)</summary>

| Check | Screenshot |
|-------|------------|
| ✅ [ChatFeedback Default](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-chatfeedback--default) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-chatfeedback--default-1.png"></a> |
| ✅ [ChatFeedback Without auto dismiss](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-chatfeedback--without-auto-dismiss) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-chatfeedback--without-auto-dismiss-2.png"></a> |
| ✅ [ChatFeedback Custom mood config](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-chatfeedback--custom-mood-config) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-chatfeedback--custom-mood-config-3.png"></a> |
| ✅ [ChatFeedback Interaction Tests - Hit Targets Resolve To The Right Mood](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-chatfeedback-chatfeedback-interaction-tests--hit-targets-resolve-to-the-right-mood) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-chatfeedback-chatfeedback-interaction-tests--hit-targets-resolve-to-the-right-mood-4.png"></a> |
| ✅ [ChatFeedback Interaction Tests - Selection Is Visible With Untintable Artwork](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-chatfeedback-chatfeedback-interaction-tests--selection-is-visible-with-untintable-artwork) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-chatfeedback-chatfeedback-interaction-tests--selection-is-visible-with-untintable-artwork-5.png"></a> |
| ✅ [ChatFeedback Interaction Tests - The Flow Advances And Returns](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-chatfeedback-chatfeedback-interaction-tests--the-flow-advances-and-returns) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-chatfeedback-chatfeedback-interaction-tests--the-flow-advances-and-returns-6.png"></a> |
| ✅ [ChatFeedback KitchenSink](https://61c19ee8d3d282003ac1d81c-lmgygghjpe.chromatic.com/?path=/story/components-kitchensink-chatfeedback--chat-feedback) | <a href="https://raw.githubusercontent.com/razorpay/blade/__ci_artifacts/artifacts/review/PR-3885/ui-critique/11-30-46/components-kitchensink-chatfeedback--chat-feedback-7.png"></a> |

</details>

### Usage

```jsx
import { ChatFeedback } from '@razorpay/blade/components';

<ChatFeedback question="How's Ray doing?" onSubmit={(payload) => console.log(payload)} onDismiss={() => setShow(false)} />
```
